quickwit-common/src/net.rs

```diff
@@ -116,6 +193,10 @@ pub fn find_available_tcp_port() -> anyhow::Result<u16> {
     Ok(port)
 }
 
+pub fn find_private_ip() -> anyhow::Result<IpAddr> {
+    unimplemented!()
```

Comment:
Let's make sure this is implemented before merging (currently PR is marked as WIP so no pb)

Reply:
I just added the implementation. Can you take a look while I finish this PR (adding comments and unit tests)?